# HG changeset patch
# User Jason Kim <jasonwkim@google.com>
# Date 1298504194 28800
# Branch pnacl-sfi
# Node ID 2c354d09798c4c01137f160a5054d94a4e0d614a
# Parent ba673e2a80bdb82072020c4d45d14024f51b5d4f
Cherrypick arm-mc-elf-r125025-lowercase.patch for .ARM.attributes. Stupid lowercase vs upppercase nonsense.

 From llvm-pnacl-0011-231-2c354d09798c4c01137f160a5054d94a4e0d614a.patch

 From .hg/patches/llvm-pnacl-0011-231-2c354d09798c4c01137f160a5054d94a4e0d614a.patch.stripped

diff -r ba673e2a80bd lib/Target/ARM/ARMAsmPrinter.cpp
--- a/lib/Target/ARM/ARMAsmPrinter.cpp	Wed Feb 23 15:36:34 2011 -0800
+++ b/lib/Target/ARM/ARMAsmPrinter.cpp	Tue Jun 07 14:28:29 2011 -0700
@@ -93,7 +93,7 @@
     void EmitTextAttribute(unsigned Attribute, StringRef String) {
       switch (Attribute) {
       case ARMBuildAttrs::CPU_name:
-        Streamer.EmitRawText(StringRef("\t.cpu ") + String);
+        Streamer.EmitRawText(StringRef("\t.cpu ") + LowercaseString(String));
         break;
       default: assert(0 && "Unsupported Text attribute in ASM Mode"); break;
       }
@@ -133,7 +133,7 @@
 
     void EmitTextAttribute(unsigned Attribute, StringRef String) {
       Contents += Attribute;
-      Contents += String;
+      Contents += UppercaseString(String);
       Contents += 0;
     }
 
@@ -556,7 +556,7 @@
 
   if (CPUString == "cortex-a8" ||
       Subtarget->isCortexA8()) {
-    AttrEmitter->EmitTextAttribute(ARMBuildAttrs::CPU_name, "CORTEX-A8");
+    AttrEmitter->EmitTextAttribute(ARMBuildAttrs::CPU_name, "cortex-a8");
     AttrEmitter->EmitAttribute(ARMBuildAttrs::CPU_arch, ARMBuildAttrs::v7);
     AttrEmitter->EmitAttribute(ARMBuildAttrs::CPU_arch_profile,
                                ARMBuildAttrs::ApplicationProfile);
